src/scribe_data/language_data_extraction/Punjabi/Shahmukhi/nouns/query_nouns.sparql

```diff
@@ -0,0 +1,45 @@
+# tool: scribe-data
+# All Shahmukhi (from Punjabi Q58635) nouns, their plurals and their genders.
+# Enter this query at https://query.wikidata.org/.
+
+# Note: We need to filter for "pnb" to select Shahmukhi words.
+
+SELECT DISTINCT
+  ?lexeme
+  (REPLACE(STR(?lexeme), "http://www.wikidata.org/entity/", "") AS ?lexemeID)
+  ?singular
+  ?plural
+  ?gender
+
+WHERE {
+  VALUES ?nounTypes {wd:Q1084 wd:Q147276} # Nouns and pronouns
+
+  ?lexeme dct:language wd:Q58635 ;   # Punjabi (Q58635)
+    wikibase:lexicalCategory ?nounTypes ;
+    wikibase:lemma ?singular .
+
+  # MARK: Plural
+
+  OPTIONAL {
+    ?lexeme ontolex:lexicalForm ?pluralForm .
+    ?pluralForm ontolex:representation ?plural ;
+      wikibase:grammaticalFeature wd:Q146786 ;
+  } .
+
+  # MARK: Gender(s)
+
+  OPTIONAL {
+    ?lexeme wdt:P5185 ?nounGender .
+    FILTER NOT EXISTS {
+      ?lexeme wdt:P31 wd:Q202444 .
+    }
+  } .
+
+  FILTER(lang(?singular) = "pnb")
```

Comment:
Doing a quick fix for this, but these filters need to be close to their selections and if there's an OPTIONAL block then they need to be inside. If they're outside the OPTIONAL, then only nouns that have plurals will be returned as those that don't will be filtered out.